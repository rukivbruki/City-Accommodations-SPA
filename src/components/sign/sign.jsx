import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {connect} from "react-redux";
import SignForm from "../sign-form/sign-form";
import PageLayout from "../page-layout/page-layout";
import Header from "../header/header";
import withInputsChange from "../../hocs/with-inputs-change/with-inputs-change";
import {getActiveCity} from "../../reducer/data/selector";

const SignFormWrapped = withInputsChange(SignForm);

const Sign = (props) => {
  const {currentCity} = props;

  return (
    <PageLayout pageClasses={[`page--gray`, `page--login`]}>

      <Header/>

      <main className="page__main page__main--login">
        <div className="page__login-container container">
          <section className="login">
            <h1 className="login__title">Sign in</h1>

            <SignFormWrapped/>

          </section>
          <section className="locations locations--login locations--current">
            <div className="locations__item">
              <Link to="/" className="locations__item-link">
                <span>{currentCity}</span>
              </Link>
            </div>
          </section>
        </div>
      </main>
    </PageLayout>
  );
};

Sign.propTypes = {
  currentCity: PropTypes.string.isRequired
};

const mapStateToProps = (state) => ({
  currentCity: getActiveCity(state),
});

export {Sign};
export default connect(mapStateToProps)(Sign);
