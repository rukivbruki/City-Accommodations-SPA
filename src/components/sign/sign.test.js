import React from "react";
import {configure, shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import {Sign} from './sign';
jest.mock(`../sign-form/sign-form.jsx`, () => jest.fn().mockReturnValue(null));

configure({adapter: new Adapter()});

it(`Sign in page renders correctly`, () => {
  const signIn = shallow(<Sign currentCity={`Paris`}/>);

  expect(signIn).toMatchSnapshot();
});
