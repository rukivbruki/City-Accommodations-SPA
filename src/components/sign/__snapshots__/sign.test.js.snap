// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in page renders correctly 1`] = `
<PageLayout
  pageClasses={
    Array [
      "page--gray",
      "page--login",
    ]
  }
>
  <Connect(Header) />
  <main
    className="page__main page__main--login"
  >
    <div
      className="page__login-container container"
    >
      <section
        className="login"
      >
        <h1
          className="login__title"
        >
          Sign in
        </h1>
        <WithInputsChange />
      </section>
      <section
        className="locations locations--login locations--current"
      >
        <div
          className="locations__item"
        >
          <Link
            className="locations__item-link"
            to="/"
          >
            <span>
              Paris
            </span>
          </Link>
        </div>
      </section>
    </div>
  </main>
</PageLayout>
`;
